x-pack/plugin/security/qa/multi-cluster/build.gradle

```diff
@@ -6,76 +6,88 @@
  */
 
 import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.internal.info.BuildParams
 import org.elasticsearch.gradle.internal.test.RestIntegTestTask
 
 apply plugin: 'elasticsearch.standalone-rest-test'
 apply plugin: 'elasticsearch.rest-resources'
 
-configurations {
-  signedCerts
-  rootCert
-}
-
-dependencies {
-  signedCerts project(path: ':x-pack:plugin:core', configuration: 'signedCerts')
-  rootCert project(path: ':x-pack:plugin:core', configuration: 'rootCert')
-}
-
```

Comment:
any reason why the existing certs are not sufficient ? Just trying to avoid more cert explosion in the source code.

Reply:
The `signedCerts`, even when they are for different clusters, are all signed by the same CA. I think this is by design for testing the trust restriction. But it does not really work here because I'd like to have completely separate trust chains for transport and remote cluster interface (and perhaps http in future). The separate trusts are necessary to test that the explicit configurations are working as intended. The `testnode` cert/key from the `core` project is often used by other test suite for configuring the transport. I could use it here as well to reduce a set of key/cert/ca. But I'd still need one set for the remote cluster. I didn't want to use the `httpCa.p12` from the core because the p12 format means the test cannot run in FIPS. It is also not really fit for the purpose since it is meant to be http. Overall, I think reusing existing certs is a bit problematic and could be confusing at times. Hence I went ahead to create dedicated ones. It's not a great solution. But it should provide better clarity till we have a better, systematic way to reuse key/cert/ca (or generate them on the fly). I hope this makes sense.